files/nrpe/check_horizon_ext

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+#
+# Check if Horizon login works externally (outside of haproxy)
+#
+# Adapted from a script by https://ask.openstack.org/en/users/17403/amedeo-salvati/
+
+STATE_OK=0
+STATE_WARNING=1
+STATE_CRITICAL=2
+STATE_UNKNOWN=3
+
+usage ()
+{
+    echo "Usage: $0 [OPTIONS]"
+    echo " -h               Get help"
+    echo " -U <username>    Username to use for Horizon login"
+    echo " -P <password>    Password to use for Horizon login"
+    echo " -R <URL>         Region URL (Keystone endpoint)"
+    echo " -H <host>        Base host part for curl call URLs"
+    echo " -I <project_id>  Project ID"
+}
+
+if (($# == 0)); then
+  usage
+  exit 1
+fi
+
+while getopts 'hU:P:R:H:I:' OPTION
+do
+    case $OPTION in
+        h)
+            usage
+            exit ${STATE_UNKNOWN}
+            ;;
+        U)
+            export HZ_TEST_USERNAME=$OPTARG
+            ;;
+        P)
+            export HZ_TEST_PASSWORD=$OPTARG
+            ;;
+        R)
+            export HZ_TEST_REGION=$OPTARG
+            ;;
+        H)
+            export HZ_TEST_HOST=$OPTARG
+            ;;
+        I)
+            export HZ_TEST_PROJECT_ID=$OPTARG
+            ;;
+	*)
+            usage
+            exit ${STATE_UNKNOWN}
+            ;;
+    esac
+done
+shift "$((OPTIND-1))"
+
+CURL="/bin/curl"
+COOKIE_FILE=~/horizontestcookie
+HORIZON_USER="$HZ_TEST_USERNAME"
+HORIZON_PASSWORD="$HZ_TEST_PASSWORD"
+HORIZON_REGION="$HZ_TEST_REGION"
+HORIZON_HOST="$HZ_TEST_HOST"
+# If project ID is omitted, the check result will be based on whichever project Horizon picks for the user.
```

Comment:
Is this true? (yes/no) I was not able to test the script in ePouta-devel

Reply:
Yup that's how it seemed to work in a limited test with my own personal non-admin account in epouta-devel. Jira task has a comment about when this should work in epouta-devel